packages/sourcecred/src/core/ledger/policies/immediate.js

```diff
@@ -48,17 +50,43 @@ export function immediateReceipts(
     );
   }
 
-  const totalIntervals = identities[0].cred.length;
-  const shortTermCredPerIdentity = identities.map(({cred}) =>
-    cred
-      .slice(
-        cred.length - Math.min(policy.numIntervalsLookback, totalIntervals)
-      )
-      .reduce((sum, cred) => sum + cred, 0)
+  const intervalsBeforeEffective = credGrainView
+    .intervals()
+    .filter((interval) => interval.endTimeMs <= effectiveTimestamp);
+
+  const numIntervalsLookback = ((
+    policy: ImmediatePolicy,
+    intervalsLength: number
+  ) => {
+    if (
+      !policy.numIntervalsLookback ||
+      policy.numIntervalsLookback > intervalsLength
+    ) {
+      return intervalsLength;
+    } else {
+      return policy.numIntervalsLookback;
+    }
+  })(policy, intervalsBeforeEffective.length);
+
+  const timeLimitedcredGrainView = credGrainView.withTimeScope(
```

Comment:
It's seeming like withTimeScope isn't a well-fitted interface for the reoccuring lookback use case. Would you make a `withLookbackScope` method in the CredGrainView class that caters to (and optimizes using slice) this lookback time scoping? That could be done in a prerequisite PR or in this one.

Reply:
Rather than making a new class, I'm inclined to just make a new way to return a TimeScopedCredGrainView by providing an effective date and a number of lookbacks. Sound good?

Reply:
..Committed for review. I prefer this approach because there is no getting around making a translation between dates and intervals. what we're provided with is an effective timestamp, so we have to parse existing intervals in the CredGrainView by timestamp any way we cut it. Would be cleaner if we could do it by slice, but I don't see a more elegant way.. Unless we want to extend CredGrainView such that intervals are indexed by their start dates, which I don't think is worth it. This at least puts it in one place that always acts the same way regardless of which policy called it.

Reply:
yep looks good. You'd have to change the way the constructor works in TimeScopedCredGrainView to make slice work. We can do that optimization later if we want.